test/integration/test_controlplane.go

```diff
@@ -392,6 +397,186 @@ func TestControlPlaneEssentials(t *testing.T) {
 	)
 }
 
+func TestControlPlaneWatchNamespaces(t *testing.T) {
+	t.Parallel()
+	namespace, cleaner := helpers.SetupTestEnv(t, GetCtx(), GetEnv())
+
+	dataplaneClient := GetClients().OperatorClient.GatewayOperatorV1beta1().DataPlanes(namespace.Name)
+	controlplaneClient := GetClients().OperatorClient.GatewayOperatorV1beta1().ControlPlanes(namespace.Name)
```

Comment:
Can't we use the controller-runtime client as we do in all our other tests? I think that sticking to a unique client improves clarity and conciseness here.

Reply:
👍 Agreed, this makes sense. f77adc903fdfe8c055ed2941bad30c1c3356c3cd